test/where-filter.test.js

```diff
@@ -42,6 +42,59 @@ context('whereFilter', () => {
 		assert.equal(array.filter(whereFilter({qty: {gte: 60}})).length, 2);
 	});
 
+	describe('wildcards and metacharacters', () => {
+		it('should match sql meta-characters', () => {
+			assert.equal(whereFilter({x: {like: 'foo%'}})({x: 'foo'}), true);
+			assert.equal(whereFilter({x: {like: 'foo%'}})({x: 'foobar'}), true);
+			assert.equal(whereFilter({x: {like: 'foo%'}})({x: 'bar'}), false);
+
+			// note that sql anchors LIKE patterns at both ^ and $, but where-filter does not
+			assert.equal(whereFilter({x: {like: 'foo%'}})({x: 'pad foobar pad'}), true);
+
+			assert.equal(whereFilter({x: {like: '%bar'}})({x: 'bar'}), true);
+			assert.equal(whereFilter({x: {like: '%bar'}})({x: 'foobar'}), true);
+			assert.equal(whereFilter({x: {like: '%bar'}})({x: 'foo'}), false);
+
+			assert.equal(whereFilter({x: {like: 'fo_'}})({x: 'foo'}), true);
+			assert.equal(whereFilter({x: {like: 'fo_'}})({x: 'foobar'}), true);
+			assert.equal(whereFilter({x: {like: 'fo_'}})({x: 'bar'}), false);
+		});
+
+		it('should match regular expression patterns', () => {
```

Comment:
Can you also add tests for something like `\\w+\.\\d'`? It seems like this still works but it's not obvious from looking at the code.

Reply:
added some PCRE tests and merged PR #28